scene/gui/rich_text_label.cpp

```diff
@@ -1467,7 +1472,7 @@ float RichTextLabel::_find_click_in_line(ItemFrame *p_frame, int p_line, const V
 			rect.size.y += get_theme_constant(SNAME("table_v_separation"));
 		}
 
-		if (p_click.y >= rect.position.y && p_click.y <= rect.position.y + rect.size.y) {
+		if (p_click.y >= rect.position.y && p_click.y <= rect.position.y + rect.size.y && p_click.x >= rect.position.x) {
 			char_pos = TS->shaped_text_hit_test_position(rid, p_click.x - rect.position.x);
```

Comment:
Probably not a good solution here, `GRAPHEME_IS_RTL` is not considered, but it would be more convenient to have one side equal to `-1` and the other equal to `glyph_count`.